modules/http2/h2_workers.c

```diff
@@ -127,10 +127,9 @@ static void wake_idle_worker(h2_workers *workers)
 {
     h2_slot *slot = pop_slot(&workers->idle);
     if (slot) {
-        int timed_out = 0;
+        int timed_out;
```

Comment:
It looks like `wake_idle_worker()` could use a loop like:
```
    for (;;) {
        slot = pop_slot());
        if (!slot) {
            if (workers->dynamic && ...) { 
                ...
            }
            break;
        }

        lock();
        timed_out = ...;
        if (!timed_out) {
            signal();
            unlock();
            break;
        }
        unlock();

        slot_done();
    }
```
rather than a recursion?

Reply:
Changed as suggested.